src/nl/hannahsten/texifyidea/grammar/LatexLexer.flex

```diff
@@ -366,4 +366,8 @@ END_PSEUDOCODE_BLOCK="\\EndFor" | "\\EndIf" | "\\EndWhile" | "\\Until" | "\\EndL
 {NORMAL_TEXT_WORD}      { return NORMAL_TEXT_WORD; }
 {NORMAL_TEXT_CHAR}      { return NORMAL_TEXT_CHAR; }
 
+// Tokens for parameter separators (e.g., \cite{param1,param2}) and
+// keyval assigns (e.g. \lstinputlisting[label=somelabel])
+"="                     { return KEYVAL_ASSIGN; }
+","                     { return PARAM_SEPARATOR; }
```

Comment:
This is indeed what I suggested, except that perhaps the name is slightly confusing because now you have keyval_assign tokens in normal text. The thing is, the token changes meaning depending on context, which is handled in the parser and not in the lexer. So perhaps `EQUALS` and `COMMA` tokens or something like that.

Reply:
Makes sense, I will change it